tuplex/core/src/physical/TransformTask.cc

```diff
@@ -19,10 +19,17 @@
 
 namespace tuplex {
     // atomic var to count output rows!
-    static std::atomic_int64_t g_totalOutputRows;
+    static std::atomic_int64_t g_totalTopOutputRows;
+    static std::atomic_int64_t g_totalBottomOutputRows;
+
+    // mapping from order number -> row count if the task is finished
+    static std::mutex g_rowsDoneMutex;
+    static std::map<size_t, size_t> g_rowsDone;
```

Comment:
to get maximum order number, do the following:
1. change to unordered map
2. Write static (!) function in TransformTask, resetLimits(max_order) which gets called and fills in  the unordered map.
3. helper function to retrieve max-task number which basically uses this preinitialized map.